iroh-net/src/discovery/pkarr.rs

```diff
@@ -91,11 +97,12 @@ impl PkarrPublisher {
     ///
     /// This is a nonblocking function, the actual update is performed in the background.
     pub fn update_addr_info(&self, info: &AddrInfo) {
-        let info = NodeInfo::new(
-            self.node_id,
-            info.relay_url.clone().map(Into::into),
-            Default::default(),
-        );
+        let (relay_url, direct_addresses) = if let Some(relay_url) = info.relay_url.as_ref() {
```

Comment:
This behavior should be documented: it publishes direct addresses if no relay URL is set, and the relay URL otherwise, but never both. Thinking more - we might want to make it an option to the `PkarrPublisher` which info to publish, and fill in the default of `RelayUrlOrDirectAdresses` or so in the iroh node builder. The latter could be a followup PR also I guess.

Reply:
can you explain the use case?